src/clf/evaluation.py

```diff
@@ -0,0 +1,186 @@
+# -*- coding: utf-8 -*-
+
+"""
+Description:
+    Comparing the performance of the different models
+
+"""
+import pandas as pd
+import numpy as np
+import imblearn
+import os
+import pickle
+from sklearn.metrics import confusion_matrix
+from sklearn.metrics import classification_report
+from sklearn.metrics import make_scorer
+from utils import load_data
+
+
+# Other metrics for model evaluation (accuracy @k optimized for and MRR @k)
+def _reciprocal_rank(true_labels: list, machine_preds: list) -> float:
+    """
+    Compute the reciprocal rank at cutoff k
+
+    Parameters
+    ----------
+    true_labels : list
+        true labels
+    machine_preds : list
+        machine predictions
+
+    Returns
+    -------
+    float
+        reciprocal rank
+    """
+
+    # add index to list only if machine predicted label exists in true labels
+    tp_pos_list = [(idx + 1) for idx, r in enumerate(machine_preds) if r in true_labels]
+
+    rr = 0
+    if len(tp_pos_list) > 0:
+        # for RR we need position of first correct item
+        first_pos_list = tp_pos_list[0]
+
+        # rr = 1/rank
+        rr = 1 / float(first_pos_list)
+
+    return rr
+
+
+def compute_mrr_at_k(items: list) -> float:
```

Comment:
Wenn der type list ist, dann am besten angeben was dort entalten ist. Also in diesem Fall list[tuples[str, str]]